src/server/worker/datum/option.go

```diff
@@ -60,6 +59,6 @@ func WithTimeout(timeout time.Duration) Option {
 // WithPrefixIndex prefixes the datum directory name (both locally and in PFS) with its index value.
 func WithPrefixIndex() Option {
 	return func(d *Datum) {
-		d.storageRoot = path.Join(d.set.storageRoot, fmt.Sprintf("%016d", d.meta.Index)+"-"+d.ID)
+		d.IDPrefix = fmt.Sprintf("%016d", d.meta.Index) + "-"
```

Comment:
What's going on here?

Reply:
No, it addresses a performance issue with the way work is laid out (we need to prefix a lexicographically ordered ID with respect to the order output by the datum iterators).